obspy/core/stream.py

```diff
@@ -3040,11 +3046,25 @@ def remove_response(self, *args, **kwargs):
             original data, use :meth:`~obspy.core.stream.Stream.copy` to create
             a copy of your stream object.
         """
-        for tr in self:
-            tr.remove_response(*args, **kwargs)
+        if nthreads > 0:
+            # multi-threaded instrument response removal. multiprocessing.dummy
+            # uses threads with the Pool API of multiprocessing
+            pool = Pool(processes=nthreads)
```

Comment:
I think you can replace the whole `if/else` with

```python
# Guard against empty Stream objects.
if not len(self):
    return self

pool = Pool(processes=max(1, min(int(nthreads), len(self.traces)))
pool.map(...)
```

The tests will catch if this does not work.